### DISCUSSION TITLE
Mapping images to product colors

### DISCUSSION THREAD

**Discussion by dohboy97** [swellstores/community#529] [Help] [answered]
Hi,

We're in the process of building a NextJs app with swell integration, similar to this one [here](https://verswell-commerce.vercel.app/product/acme-bomber-jacket). 

However, unlike the linked app I want functionality where toggling a color updates the selected image to match the toggled color. How does the swell team suggest mapping the `images` on a product to the color `option` on a product? 

I see that maybe we could add `images` to the `values` fields on the color option? It'll take a bit of time to get those added so wanted to see if there was another recommended path forward.

Here's an example of what our color option values look like now:
```
{
              "id": "123456789",
              "name": "Black",
              "price": null,
              "shipment_weight": null,
              "description": "",
              "hex_codes": "#000000"
            },
```

Thanks,
Connor

**Comment by logeshswell** [comment, marked as answer]
You can try adding an image to the [variant](https://developers.swell.is/backend-api/variants) field inside the products model.
Would that work for you?
